### ISSUE TITLE
[Pipeline] Tests run twice on Pull Request Creation

### ISSUE THREAD

**Issue by jlprat** [aiven/karapace#271]
# What happened?
When a new pull request is created the test suite is run twice:
- One because it's triggered at pull request creation
- One because there is a push event

# What did you expect to happen?
The test suite is run only once when the following happens:
- Pull request creation (pr creation + push)
- Pushing more than one commit at a time
- Re opening the pull request

**Comment by jlprat** [comment]
Actually, Linting also suffers from the same thing
